docs/content/en/docs/reference/clusterspec/cloudstack.md

```diff
@@ -5,4 +5,326 @@ weight: 30
 description: >
   Full EKS Anywhere configuration reference for a CloudStack cluster.
 ---
+This is a generic template with detailed descriptions below for reference.
+The following additional optional configuration can also be included:
 
+* [CNI]({{< relref "optional/cni.md" >}})
+* [IAM for pods]({{< relref "optional/irsa.md" >}})
+* [IAM Authenticator]({{< relref "optional/iamauth.md" >}})
+* [OIDC]({{< relref "optional/oidc.md" >}})
+* [gitops]({{< relref "optional/gitops.md" >}})
+* [proxy]({{< relref "optional/proxy.md" >}})
+* [Registry Mirror]({{< relref "optional/registrymirror.md" >}})
+
+
+```yaml
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
+metadata:
+  name: my-cluster-name
+spec:
+  clusterNetwork:
+    cniConfig:
+      cilium: {}
+    pods:
+      cidrBlocks:
+      - 192.168.0.0/16
+    services:
+      cidrBlocks:
+      - 10.96.0.0/12
+  controlPlaneConfiguration:
+    count: 3
+    endpoint:
+      host: ""
+    machineGroupRef:
+      kind: CloudStackMachineConfig
+      name: my-cluster-name-cp
+    taints:
+    - key: ""
+      value: ""
+      effect: ""
+    labels:
+      "<key1>": ""
+      "<key2>": ""
+  datacenterRef:
+    kind: CloudStackDatacenterConfig
+    name: my-cluster-name
+  externalEtcdConfiguration:
+    count: 3
+    machineGroupRef:
+      kind: CloudStackMachineConfig
+      name: my-cluster-name-etcd
+  kubernetesVersion: "1.23"
+  managementCluster:
+    name: my-cluster-name
+  workerNodeGroupConfigurations:
+  - count: 2
+    machineGroupRef:
+      kind: CloudStackMachineConfig
+      name: my-cluster-name
+    taints:
+    - key: ""
+      value: ""
+      effect: ""
+    labels:
+      "<key1>": ""
+      "<key2>": ""
```

Comment:
A special label value is supported by CAPC provider: 

```
labels:
      cluster.x-k8s.io/failure-domain: ds.meta_data.failuredomain
```
`ds.meta_data.failuredomain` will be replaced with a failuredomain name where the node is deployed in, for example `az-1`

Reply:
Done.